File: FoodDelivery.DAL/EF/Entities/Restaurant.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace FoodDelivery.DAL.EF.Entities
{
	public class Restaurant
	{
		[Key]
		public Guid Id { get; set; }

		[ForeignKey("Owner")]
		public Guid OwnerId { get; set; }

		public string Name { get; set; }

		public string Description { get; set; }

		public int Status { get; set; }

		public Guid? ImageName { get; set; }

		[ForeignKey("Type")]
		public int TypeId { get; set; }

		public virtual UserProfile Owner { get; set; }
		public virtual RestaurantType Type { get; set; }
		public virtual List<RestaurantAddress> Addresses { get; set; }
		public virtual List<Dish> Dishes { get; set; }
		public virtual List<RestaurantRating> Ratings { get; set; }
		public virtual List<RestaurantOrder> RestaurantOrders { get; set; }
	}
}

File: FoodDelivery.Web/src/app/models/restaurant/restaurant.ts
import { dish } from 'src/app/models/dish/dish';
import { Guid } from 'guid-typescript';
import { restaurantType } from './restaurantType';
import { restaurantAddress } from './restaurantAddress';

export class restaurant{
    public id?: Guid;
    public ownerId: Guid;
    public name: string;
    public description: string;
    public rating?: number;
    public type: restaurantType;
    public addresses?: restaurantAddress[];
    public dishes?: dish[];
}